### ISSUE TITLE
[Class] [Warrior] [Critical Bug] - Protection & Beserker Stance Bug

### ISSUE THREAD

**Issue by tk808** [dalaranwow/dalaran-wow#4365]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Gameserver.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

**Current behaviour**: Tell us what happens in detail (Where, When, What, Why, Who)

This is a compilation of Warrior Stances bugging, which at times is very frustrating, and at various random points in raids (mainly tested in TOGC). I've looked through dalaran forums and the git, but I've not seen anyone post this issue. But within game, almost every high-end warrior has experienced this issue, and occasionally can be resolved by doing a (stance swap), in or before combat, but not all the time it resolves the issue.  

In Summary, stance bugging refers to the lose of all modifiers pertaining to their spec, in that specific stance. Warriors by nature have everything baked into Protection, Beserker or Arms stances, thus when they become "bugged" it becomes very apparent that there is significant loss of damage (fury) or significant intake of damage (protection). 

1) Among Protection Warriors (Protection Stance), we realize that we are: taking extra damage (usually double), and occasionally losing all threat modifiers (45%) baked directly into protection stance. 

2) Among Fury warriors (Beserker Stance), when zerker becomes bugged, warriors lose the damage modifiers and the hard-lining 3% critical damage, and all the damage modifiers baked into Beserker stance.

**Expected behavior**: Tell us what should happen instead

The stances should not be bugging at any point. On warmane and many other WOTLK private servers, the same issue occurs, with stances bugging.

**Steps to reproduce the problem**:

Note: This is a very critical problem, that can't be re-produced very easily. Due to the randomness, it's either a class spell is interacting with stances, or a boss encounter is interacting with warrior stances. 

Now, transmogging warriors, battle-resing, hand of protection, and anything that changes the warriors appearances sometimes can active the loss of the modifiers of stances (before combat), as shown in the video below. But sometimes the loss of stance modifers, pertain even after stance swapping, after receiving an effect.  

To reproduce for protection, Gormok the impaler (TOGC 25) is the only boss in the current content where this disparity among tanks becomes of hot debate

Protection Warrior:

1. Enter TOGC 25 with protection warrior
2. Go through the entire gormok fight while stacking at least 4-5 impale stacks, then drop them and repeat until gormok anger rises (so that he increases damage). Test the damage intake through 2-4 anger cycles. 
3. Die after u get 4 impale stacks (and get a battle res mid-combat, or take a bop, or etc), and resume gaining stacks of impale.
4. Test again with prot warrior without resses, occasionally warrior will take more damage than prot pally, even without stance bug, which is incorrect. with prot pally (without ardent defender). Prot warriors and prot pallys should approximately take the same amount of damage, while a protection warrior full buffed should take less damage every time than a prot pally. by approximately 1%.

If the stance is indeed bugged, prot warrior will have taken roughly double the damage of a prot pally, when the stance is bugged. 

**Include proofs for this behaviour**

Prot pally tank with 7 impale stacks, i get bopped, i have 0 stacks and get 1 shot. As seen in this video. Note: ardent defender only procs once, and not the bosses anger (which is nearly at 5) The most definitive proof of this major issue (fyi i have approximately 56k HP fully buffed with like 5450 GS, prot pally has around 5200 GS with 53k hp fully buffed.

I took a 58k Impale hit on anger 5, with 0 impale stacks on me. Now, every tank should have a CD for every impale, but getting 1 shot on 0 impales is a non-issue for dks,prot pally, druid. The conclusion we come to here is it's the warrior's defensive stance mitgation

I want it to be known that this creates a stigma for protection warriors within the server. Warrior is the most played class in WOW, and most likely on DAL. 

https://clips.twitch.tv/cubsbum/LovelySquirrelSmoocherZ

Please also reference, which may coincide with this issue: https://github.com/dalaranwow/dalaran-wow/issues/1510

**Include the ID for the game objects, npcs (creatures,pets,minions), spells, items, quests, instances, zones, achievements, skills**

1) Warrior - Protection Stance
2) Warrior - Beserker Stance

**Include Screenshots from the issue if necessary**

https://cdn.discordapp.com/attachments/240972546428764170/276075573523251200/WoWScrnShot_013017_182403.jpg

Thank you for addressing this issue,
Thygore

**Comment by Domynator** [comment]
I am providing more images of the stance bug, this bug is also in Fury.

When switching around gear, I lost my 2% increase from the set bonus (chest/helm) and also the 5% from fury stance.

See attached images, see the critical chance.
![asdf1](https://cloud.githubusercontent.com/assets/25484076/22511825/85a5175e-e85c-11e6-8599-ccc4975e4e60.png)
![asdf2](https://cloud.githubusercontent.com/assets/25484076/22511828/872490c8-e85c-11e6-8fe2-d7c228748348.png)
![asdf3](https://cloud.githubusercontent.com/assets/25484076/22511831/8bac809c-e85c-11e6-82d6-400b59546b9b.png)

-Domynator

**Comment by tk808** [comment]
This issue regarding warriors doesn't only pertain to Gormok TOGC, it happens at all levels, and in practically any scenario. The reason we've referenced TOGC, is because it requires everyone to be at nearly their best, and any under-performing of players, classes/specs becomes highly recognizable that something is not correct with the server. 

I've outlined in the post above, reasons why this issue occurs for warriors. 

Once again, due to the hype of impending ICC release, i don't want this to be an issue about TOGC raid, when in fact it's an inherent issue anywhere, anytime, with warriors.

**Comment by Mike-Hunt** [comment]
@tk808 You say "If the stance is indeed bugged, prot warrior will have taken roughly double the damage of a prot pally", what causes the double damage? Shouldn't the only damage taken modifiers missing be 10% Damage Reduction via Defensive Stance and 6% spell damage reduction via Improved Defensive Stance? Note that the 10% damage reduction would've been the difference between you getting oneshot and not getting oneshot on that screenshot so it's still definitely significant, I'm just not sure how you reached the conclusion of double damage.

@Domynator I don't see how the images you provide show proof of Berserker Stance bugging. 
In the first screenshot you are in Berserker Stance without Rampage at 41.17% crit chance.
Second screenshot in Battle Stance without Rampage at 36.17% crit chance (5% lower because missing 3% from Berserker Stance + 2% from 2T9 bonus)
Third screenshot in Berserker Stance with Rampage at 46.17% crit chance, which is 5% higher than the first screenshot due to Rampage. 
All of the stances appear to be working, the only buggy thing is the first screenshot where you are presumably in a Fury spec without having a Rampage buff. The changes in Attack Power are due to the loss of Devotion Aura combined with Armored to the Teeth.

If anything, it seems to show #2353 one of the bugs where you can lose Rampage, something like "Rampage breaks even if the warrior is alone and respecs into Rampage. Gotta relog to get the effect"

**Comment by Domynator** [comment]
@Mike-Hunt you say that you dont see the problem, look again. The gear is the same, the stances are correct, I lost 10% crit on the image with 36% crit when I normally have 46% crit. 

This occured when I switched gear around, and also it occurs when fighting bosses such as Anub and Yogg.

Many times ive come into a fight, from a previous fight, and not be able to do more than 3-4k dps.

I am 5.3 gs fury warrior who on the normal can pull 7k single target without trying.

There is  a bug, please dont comment unless you have USEFUL information. 

Trolling a github post for a real issue.... Man this is the future isnt it?

**Comment by tk808** [comment]
@Mike-Hunt, It's roughly double the damage, because when you lose protection stance buff, the difference between one shout and survival is huge. The damage increases from Gormok impaler (anger), the damage is no linear, in that case your argument would be valid.

If you have evidence countering the arguing the argument of double damage from gormok, by all means provide it. 

And regarding fury warriors, you need to look @ this pictures further.

Despite what any nay-sayer would like to add in these discussions, the issue is more relevant than ever due to stances bugging.

**Comment by filthymonkey** [comment]
I too experience this bug as a fury warrior. Not only does the crit % drop, dps and physical damage modifiers drop as well. Titan's Grip drops physical damage by 10% therefore your physical damage is at x90% un-buffed and roughly x93% buffed. however, when the bug occurs physical damage is reduced to x85%. Activating off spec talents and repeatedly changing stances along with a relog seems to fix it temporarily.

**Comment by tk808** [comment]
`Here is further proof of BOP kicking you out of defensive stance`

View weapon damage (should be negated by 5%): 

https://cdn.discordapp.com/attachments/240972546428764170/276568704513343488/WoWScrnShot_020117_202427.jpg 

https://cdn.discordapp.com/attachments/240972546428764170/276568732115927050/WoWScrnShot_020117_202432.jpg

**Comment by Mike-Hunt** [comment]
@Domynator I'm just trying to understand because it is a quite weird bug. 
But like I said, in the image where you have 36% crit you have no Rampage buff and you are in Battle Stance so you are missing 5% (Rampage) + 3% (Berserker Stance) + 2% (2T9 bonus to Berserker Stance) compared to the image with 46% crit. Similarly, the 5% difference in crit between the first image and the third image is caused by the loss of the Rampage aura. I'm not "trolling your github post", I'm seeking clarification since from what I can tell the difference in crit chance is explainable without any bug in stances.

**Comment by strobe-light** [comment]
Alright so I have done some tests as well. With what seems to be a 100% reproduction rate, every time a protection warrior gets Hand of Protection, they lose: 
-5% damage done 
-10% damage taken
+% threat generation
from Defensive Stance until they relog or shift stances. The -6% spell damage taken from Improved Defensive Stance seems to be unaffected.
The damage done/taken were tested with Consecration and Thunderclap without any proc items as these abilities have no variance in damage as long as the AP/SP and modifiers are unchanged. The threat generation was tested with Sunder Armor + a stopattack macro.
Before HoP was applied, my Thunderclap was doing 205 damage per hit, Consecration was hitting me for 198 per tick, and Sunder Armor was doing ~1000 threat per application. 
After HoP was applied, my Thunderclap was doing 216 damage per hit, Consecration was hitting me for 220 per tick and Sunder Armor was doing ~600 threat per application.
216 x 0.95 = 205. 220 x 0.9 = 198. Not sure about the threat calculation, as far as I know Defensive Stance should cause a 45% increase in threat but 600 x 1.45 = ~870 which is a bit off 1000.
http://imgur.com/a/EAYvd

As fury, every time I got Hand of Protection I lost the 10% penalty from Titan's Grip, but retained the 20% strength boost from Improved Berserker Stance and the 3% crit chance boost from Berserker Stance. I could change stances back and forth and I still didn't get the 10% penalty from Titan's Grip until I re-equipped my weapons, which I guess makes sense since Titan's Grip's penalty isn't tied to any particular stance.
http://imgur.com/a/LYtAd

I was unable to reproduce losing the 20% strength boost or 3% crit chance boost mentioned by @Domynator, his screenshots all look like Rampage bug as well. Could be that something else can cause stance bugging for fury though, I didn't test (battle) rezzes/Savory Deviate Delights/phasing into instances to name a few.

tl;dr version: hop clears defensive stance modifiers and titan's grip penalty, doesn't seem to remove crit or +20% strength boost from berserker stance.

**Comment by filthymonkey** [comment]
Bump

**Comment by strobe-light** [comment]
@filthymonkey Bumping doesn't actually help, everyone can see that this issue is open with a 'critical priority' tag. 
If you want to contribute, do some tests and find a way to reproduce what you mention in your statement 

> I too experience this bug as a fury warrior. Not only does the crit % drop, dps and physical damage modifiers drop as well. Titan's Grip drops physical damage by 10% therefore your physical damage is at x90% un-buffed and roughly x93% buffed. however, when the bug occurs physical damage is reduced to x85%. 

Since there is no proof of that so far in this thread.

**Comment by Edzero** [comment]
I am unsure if #1510 will affect this. 

I will test them on the PTR when I have the opportunity to provide a more complete report.

**Comment by filthymonkey** [comment]
![wowscrnshot_020517_013229](https://cloud.githubusercontent.com/assets/25497726/22624263/4747ab3a-eb44-11e6-927e-e466323bdc45.jpg)
@strobe-light hows this?

**Comment by filthymonkey** [comment]
![wowscrnshot_020517_020835](https://cloud.githubusercontent.com/assets/25497726/22624372/5a21cbf6-eb48-11e6-969b-3a25f2fc0409.jpg)
and some more

**Comment by strobe-light** [comment]
The screenshots aren't that useful by themselves without context, but if you 

- identify the cause that makes you get to that state and see if it can be reliably reproduced. Is it just dying and zoning in to an instance that does it? Since it is Anub, was HoP involved at all? Could you just stance dance and fix it?
- determine if your strength and/or crit chance change when you're in that state independent of other buffs (kings/motw etc) or gear changes to demonstrate if the +20% bonus str and/or 3% crit are also affected

then yes that's a lot more helpful.

**Comment by filthymonkey** [comment]
yeah sorry it was like 2 A.M.
This happened after receiving BoP. after bop dropped i received 0 damage reduction (x100%). After dying and coming back into the instance this is what i got. All buffs worked properly but was still left with the insane damage reduction.

**Comment by Dartheus** [comment]
I sort of have a feeling this might be the case why i sometimes get oneshot out of nowhere by gormok whilst other times i have no issues :S I will watch out for this and post further proof if i find something :)

**Comment by Rushor** [comment]
the reason for this issue is http://wotlk.openwow.com/spell=7381

you get http://wotlk.openwow.com/spell=7381 by default if you log in and it does not appear in the log.

if you cast then http://wotlk.openwow.com/spell=2458, you get another stack of aura http://wotlk.openwow.com/spell=7381.

but you should always only have one stack of http://wotlk.openwow.com/spell=7381.

**Comment by Rushor** [comment]
Scripts/Spells: Warrior - Remove Berserker Stance Passive if Berserker Stance is not active 

* closes: https://github.com/dalaranwow/dalaran-wow/issues/4365
* commit: 15daaa40ff16b9564ff77b04662a7d9057b39b0b

The issue was fixed by a developer and will now be tested on the Public Test Realm (PTR). The changes/fixes will be applied with the NEXT changelog on the liverealm.
This means that the issue exists in game at the moment i wrote this message.
Only once the issue was marked with "Fixed - On Live Server", you can find the fix on you server.
Please watch http://www.dalaran-wow.com/news/ to see when the changelog and the fix was applied.

**Comment by Rushor** [comment]
so the issue was fixed now, you will no longer get two stacks of http://wotlk.openwow.com/spell=7381 and the damage don/taken should be normal again.  The fix will be tested on PTR once it was updated. Thanks guys for all the help.

**Comment by plznoz** [comment]
Bit confused by that explanation @Rushor.
1. Defensive stance gets broken by HoP (lose -dmg taken, -dmg done, +threat)
2. Fury warrior's Titan's Grip penalty gets removed by HoP (#1065)

(See @strobe-light's comment made on Feb 4)
Are both of these fixed by that commit and (soon) on ptr™?
Cheers

**Comment by Rushor** [comment]
@plznoz the loose of Fury warrior's Titan's Grip penalty/Defensive stance by HoP is another issue too - i will fix it as fast as i can

**Comment by Rushor** [comment]
hm actually this already works on PTR:

defensive stance aura list before HoP

```sql
[19:53:33]id: 71 [Defensive Stance] effmask: 1 charges: 0 stack: 1 slot 0 duration: -1 maxduration: -1   caster: player guid: 43
[19:53:33]id: 5301 [Defensive State (DND)] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[19:53:33]id: 7376 [Defensive Stance Passive] effmask: 7 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[19:53:33]id: 12678 [Stance Mastery] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[19:53:33]id: 12886 [Improved Challenging Shout] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[19:53:33]id: 20582 [Quickness] effmask: 3 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[19:53:33]id: 20583 [Nature Resistance] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[19:53:33]id: 21009 [Elusiveness] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[19:53:33]id: 32215 [Victorious State] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[19:53:33]id: 45471 [Defiance Expertise Passive (DND)] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[19:53:33]id: 57339 [Tank Class Passive Threat] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
```

defensive stance aura list after HoP cast on warrior:

```sql
[19:54:53]id: 71 [Defensive Stance] effmask: 1 charges: 0 stack: 1 slot 0 duration: -1 maxduration: -1   caster: player guid: 43
[19:54:53]id: 5301 [Defensive State (DND)] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[19:54:53]id: 7376 [Defensive Stance Passive] effmask: 7 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[19:54:53]id: 12678 [Stance Mastery] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[19:54:53]id: 12886 [Improved Challenging Shout] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[19:54:53]id: 20582 [Quickness] effmask: 3 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[19:54:53]id: 20583 [Nature Resistance] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[19:54:53]id: 21009 [Elusiveness] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[19:54:53]id: 25771 [Forbearance] effmask: 1 charges: 0 stack: 1 slot 2 duration: 107603 maxduration: 120000   caster: player guid: 42
[19:54:53]id: 32215 [Victorious State] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[19:54:53]id: 45471 [Defiance Expertise Passive (DND)] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[19:54:53]id: 57339 [Tank Class Passive Threat] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[19:54:53]id: 61987 [Avenging Wrath Marker] effmask: 1 charges: 0 stack: 1 slot 3 duration: 17603 maxduration: 30000   caster: player guid: 43
[19:54:53]id: 61988 [] effmask: 1 charges: 0 stack: 1 slot 4 duration: 167603 maxduration: 180000   caster: player guid: 42
```

warrior aura list with Titan's Grip penalty before HoP cast on warrior:

```sql

[20:03:37]id: 71 [Defensive Stance] effmask: 1 charges: 0 stack: 1 slot 0 duration: -1 maxduration: -1   caster: player guid: 43
[20:03:37]id: 5301 [Defensive State (DND)] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[20:03:37]id: 7376 [Defensive Stance Passive] effmask: 7 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[20:03:37]id: 12678 [Stance Mastery] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[20:03:37]id: 12835 [Booming Voice] effmask: 3 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:03:37]id: 12853 [Cruelty] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:03:37]id: 12858 [Commanding Presence] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:03:37]id: 12876 [Improved Demoralizing Shout] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:03:37]id: 12886 [Improved Challenging Shout] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[20:03:37]id: 12972 [Flurry] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:03:37]id: 13000 [Unbridled Wrath] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:03:37]id: 13046 [Enrage] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:03:37]id: 16489 [Blood Craze] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:03:37]id: 20501 [Improved Berserker Rage] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:03:37]id: 20503 [Improved Execute] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:03:37]id: 20582 [Quickness] effmask: 3 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[20:03:37]id: 20583 [Nature Resistance] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[20:03:37]id: 21009 [Elusiveness] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[20:03:37]id: 29592 [Precision] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:03:37]id: 29776 [Improved Whirlwind] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:03:37]id: 29801 [Rampage] effmask: 1 charges: 0 stack: 1 slot 1 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:03:37]id: 29889 [Improved Intercept] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:03:37]id: 32215 [Victorious State] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[20:03:37]id: 45471 [Defiance Expertise Passive (DND)] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[20:03:37]id: 46911 [Furious Attacks] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:03:37]id: 46915 [Bloodsurge] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:03:37]id: 49152 [Titan's Grip] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[20:03:37]id: 56932 [Unending Fury] effmask: 2 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:03:37]id: 57339 [Tank Class Passive Threat] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
```


warrior aura list with Titan's Grip penalty after HoP cast on warrior:

```sql
[20:04:42]id: 71 [Defensive Stance] effmask: 1 charges: 0 stack: 1 slot 0 duration: -1 maxduration: -1   caster: player guid: 43
[20:04:42]id: 5301 [Defensive State (DND)] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[20:04:42]id: 7376 [Defensive Stance Passive] effmask: 7 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[20:04:42]id: 12678 [Stance Mastery] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[20:04:42]id: 12835 [Booming Voice] effmask: 3 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:04:42]id: 12853 [Cruelty] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:04:42]id: 12858 [Commanding Presence] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:04:42]id: 12876 [Improved Demoralizing Shout] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:04:42]id: 12886 [Improved Challenging Shout] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[20:04:42]id: 12972 [Flurry] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:04:42]id: 13000 [Unbridled Wrath] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:04:42]id: 13046 [Enrage] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:04:42]id: 16489 [Blood Craze] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:04:42]id: 20501 [Improved Berserker Rage] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:04:42]id: 20503 [Improved Execute] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:04:42]id: 20582 [Quickness] effmask: 3 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[20:04:42]id: 20583 [Nature Resistance] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[20:04:42]id: 21009 [Elusiveness] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[20:04:42]id: 25771 [Forbearance] effmask: 1 charges: 0 stack: 1 slot 3 duration: 102723 maxduration: 120000   caster: player guid: 42
[20:04:42]id: 29592 [Precision] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:04:42]id: 29776 [Improved Whirlwind] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:04:42]id: 29801 [Rampage] effmask: 1 charges: 0 stack: 1 slot 1 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:04:42]id: 29889 [Improved Intercept] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:04:42]id: 32215 [Victorious State] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[20:04:42]id: 45471 [Defiance Expertise Passive (DND)] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[20:04:42]id: 46911 [Furious Attacks] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:04:42]id: 46915 [Bloodsurge] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:04:42]id: 49152 [Titan's Grip] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[20:04:42]id: 56932 [Unending Fury] effmask: 2 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  [talent] caster: player guid: 43
[20:04:42]id: 57339 [Tank Class Passive Threat] effmask: 1 charges: 0 stack: 1 slot 255 duration: -1 maxduration: -1  [passive]  caster: player guid: 43
[20:04:42]id: 61987 [Avenging Wrath Marker] effmask: 1 charges: 0 stack: 1 slot 4 duration: 12723 maxduration: 30000   caster: player guid: 43
[20:04:42]id: 61988 [] effmask: 1 charges: 0 stack: 1 slot 5 duration: 162723 maxduration: 180000   caster: player guid: 42
```

So the warrior does not loose any auras here. I think that got fixed by

Core/Spell: revamp of immunities system

- Spell attributes research
- Container optimization
- Do not skip damage immunity checks on SPELL_ATTR0_UNAFFECTED_BY_INVULNERABILITY present
- Added additonal arguments support to SendCastResult / CheckCast, spells can now send proper error messages
- Loads spell immune masks/states/effects at startup.
- This cached info is used whenever a spell cast requires immunities to be known beforehand (to see if an aura could be applied for example)

Closes #16658
Closes #16901
Closes #18114

* commit: 50a3ce5703bdaa4fe16e2e773eabafa9a4818083

Core/Spell: Improve immunity system.

Ref: https://github.com/TrinityCore/TrinityCore/commit/425f6c3c34b72190e481ba89fd4757df238475d8
https://github.com/TrinityCore/TrinityCore/commit/f3bdd705c0e3ab28585406ae8e08da44a09b14df

* commit: 654739dc421236cdd1815746ce8cd1ba3cfb4236

Those fixes are on PTR and needs to be tested.

**Comment by tk808** [comment]
@Rushor 

Excellent Job Rushor. Thank you, and everyone who contributed to identify this issue, in a relatively short period of time! Everything now makes sense for the reason why protection and berserk stance bugs out.

Cheers,
Thygore

**Comment by ashido96** [comment]
This still sometimes happens to warrios in ICC, what needs to be done to get stuff applied, like is it not tested? How can they be tested? If I test them then will they applied to the server (or just keep sitting here)?

**Comment by plznoz** [comment]
Considering Teridan and Ed has tested a lot of things which haven't been applied I think the problem lays somewhere else...

**Comment by robinsch** [comment]
Too much text. I fixed Defensive Stance Passive and Titan's Grip removal.

@TeridanDW Could you sum up what's still missing in this report?

**Comment by TeridanDW** [comment]
> Defensive Stance

Battle and Berserker stance passives been removed by hop also and the commits from @Rushor about the immunity system.

**Comment by robinsch** [comment]
We will skip the immunity system at first.

**Comment by TeridanDW** [comment]
Berserket stance passives are still an issue on ptr.
After character reloged:
![image](https://user-images.githubusercontent.com/6797611/32046772-9ea0b3e0-ba4c-11e7-82f3-da12cae324cb.png)

After hop and change to deffensive stance and back to berserker stance (which can't happen, notice the error message when i spem berserker stance):
![image](https://user-images.githubusercontent.com/6797611/32046869-ecb6fe40-ba4c-11e7-94f0-99c00e0450ac.png)
